certora/specs/AccessControlDefaultAdminRules.spec

```diff
@@ -203,6 +269,24 @@ rule beginDefaultAdminTransfer(env e, address newAdmin) {
     "pending default admin delay is set";
 }
 
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Rule: A default admin can't change in less than the applied schedule                                                │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule pendingDefaultAdminDelayEnforced(env e1, env e2, method f, calldataarg args, address newAdmin) {
```

Comment:
Seeing this rule now I'm realizing that my initial suggestion didn't make a lot of sense. This rule doesn't convince me that there is no way to change the admin faster, because it's only looking at ways that consist of invoking a single function (`f`). What if there was another way to violate the property?

I have some ideas for how we might try to prove that using an invariant and a ghost variable, but I don't know if they make sense fully...

We can just leave this rule as is with a comment saying that there may be ways to generalize it.

Reply:
My answer to this is basically [this comment](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4180#discussion_r1184005586). TLDR; is that generalizing the rule detaches the change from the `delayBefore`.